### PR TITLE
Eh3rrera c c++ functional style programming using c

### PR DESCRIPTION
**Pull request by eh3rrera**
Corrected code formatting and grammar and typo fixes.

### CODE REVIEW

**Comment by durden** [comment]
Thanks @eh3rrera! Any thoughts on these suggestions @phaysaal?  I can merge them if you want or let me know what you want to merge specifically.  You can also edit your guide directly, but merging will give @eh3rrera github contribution 'credit'.

File: published/c-c++/functional-style-programming-using-c/article.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
-> LEARNING PROGRAMMING AND LEARNING PROGRAMMING LANGUAGE ARE NOT THE SAME. MOST OF OUR TEXTBOOKS TEACH US PROGRAMMING LANGUAGES. IN TRUTH, A PROGRAMMING LANGUAGE IS JUST A TOOL THAT WE USE TO IMPLEMENT AN EXECUTABLE PROGRAM. WE NEED RESOURCES THAT TEACH US HOW TO _SOLVE PROBLEMS_ USING A PROGRAMMING LANGUAGE.
+> Learning programming and learning programming language are not the same. most of our textbooks teach us programming languages. In truth, a programming language is just a tool that we use to implement an executable program. We need resources that teach us how to _solve problems_ using a programming language.
```
Comment (phaysaal) [inline published/c-c++/functional-style-programming-using-c/article.md:1]: _solve problems_ 
can be substituted by 
"solve problems" 
or simply
solve problems

**Comment by phaysaal** [comment]
@durden, I think the changes are OK. So you can merge them. @eh3rrera , thanks for working on it. Any thoughts on extension with new sections?

**Comment by eh3rrera** [comment]
Thanks @durden!

@phaysaal Maybe some sections about how to implement functions like filter, map, foldLeft, etc. Another section could be what a closure is and how to implement it in C. Anyway, thanks for writing your guide.

**Pull request merged**